torch/csrc/jit/python/pybind_utils.cpp

```diff
@@ -874,10 +874,19 @@ std::optional<py::object> _maybe_handle_torch_function(
   std::vector<PyObject*> overloaded_args;
   const auto args_size = args.size();
   size_t total_arg_num = args_size + kwargs.size();
+  PyObject* const args_ptr = args.ptr();
   for (const auto i : c10::irange(args_size)) {
-    is_tensor_and_append_overloaded(args[i].ptr(), &overloaded_args);
+    // Because pybind object indexing is implemented generically for
+    // all objects, operator[] returns py::object instead of
+    // py::handle, so args[i].ptr() would cause a reference count
+    // round trip. This has enough overhead that I noticed it while
+    // profiling and came here to fix it. For tuples in particular,
+    // PyTuple_GET_ITEM returns a borrowed reference, so no counting
+    // overhead.
+    auto* const args_i_ptr = PyTuple_GET_ITEM(args_ptr, i);
```

Comment:
Please use PyTuple_GetItem for proper bound checking.

Reply:
we know that the argument is in range because `i` is iterating over `args.size()`. Do you mean that we should ignore this and bounds check anyway?

Reply:
> if we consider this to be perf-critical, I'm ok with skipping the bound check.

I don't think it's perf-critical, just noticeable. I'll change it to PyTuple_GetItem and leave a note that the bounds check could be skipped if anyone comes back and looks at the code because of perf.